client/blocks/login/social.jsx

```diff
@@ -106,6 +106,50 @@ class SocialLoginForm extends Component {
 		);
 	};
 
+	handleAppleResponse = response => {
+		const { onSuccess, redirectTo } = this.props;
+
+		if ( ! response.id_token ) {
+			return;
+		}
+
+		const user = response.user || {};
+
+		const socialInfo = {
+			service: 'apple',
+			id_token: response.id_token,
+			user_name: user.name,
+			user_email: user.email,
+		};
+
+		this.props.loginSocialUser( socialInfo, redirectTo ).then(
+			() => {
+				this.recordEvent( 'calypso_login_social_login_success', 'apple' );
+
+				onSuccess();
+			},
+			error => {
+				if ( error.code === 'unknown_user' ) {
+					return this.props.createSocialUser( socialInfo, 'login' ).then(
+						() => this.recordEvent( 'calypso_login_social_signup_success', 'apple' ),
+						createAccountError =>
+							this.recordEvent( 'calypso_login_social_signup_failure', 'apple', {
+								error_code: createAccountError.code,
+								error_message: createAccountError.message,
+							} )
+					);
+				} else if ( error.code === 'user_exists' ) {
+					this.props.createSocialUserFailed( 'apple', response.id_token, error );
```

Comment:
It looks like we got it wrong here and for Google since this `createSocialUserFailed()` action only accepts [two parameters](https://github.com/Automattic/wp-calypso/blob/3dc5e7f41e1ba6ad2fb2e42cc9858e9e72f6c119/client/state/login/actions.js#L397).

Reply:
Indeed its signature was updated there https://github.com/Automattic/wp-calypso/pull/17208/files#diff-ea69383340d9783c30ab56da9972fabeR344 but not all calls reflected it